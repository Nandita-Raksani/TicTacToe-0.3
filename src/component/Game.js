import React from 'react';
import '../App.css';
import StyleConstants from '../constants/StyleConstants';

const Game = () => {
    return (
        <div className={StyleConstants.APP}>
            <header className={StyleConstants.APP_HEADER}>
                {StyleConstants.APP_TITLE}
            </header>
        </div>
    );
}
export default Game;