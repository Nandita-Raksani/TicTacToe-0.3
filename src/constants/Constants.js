const Constants = {
    MAXIMUM_NUMBER_OF_TILES: 9,
    INITIAL_TILE_POSITION: 0,
    PLAYER_X: "X",
    EMPTY_VALUE: "",
    PLAYER_O: "O",
    CURRENT_PLAYER: "Current Player : ",
    FIRST_ROW_TILES: [0, 1, 2],
    SECOND_ROW_TILES: [3, 4, 5],
    WINNER: "Winner is : ",
    THIRD_ROW_TILES: [6, 7, 8],
    FIRST_COLUMN_TILES: [0, 3, 6],
    SECOND_COLUMN_TILES: [1, 4, 7],
    THIRD_COLUMN_TILES: [2, 5, 8],
    LEFT_DIAGONAL_TILES: [0, 4, 8],
    RIGHT_DIAGONAL_TILES: [2, 4, 6]
};
export default Constants;