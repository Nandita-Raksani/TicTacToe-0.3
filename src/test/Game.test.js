import React from 'react';
import Game from '../component/Game';
import Constants from './constants/Constants';
import { shallow } from 'enzyme';

describe(("<Game/> component"), () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<Game />);
    });

    it("should render correctly", () => {
        expect(wrapper).toMatchSnapshot();
    });

    it("should render title correctly", () => {
        expect(wrapper.find("header").text()).toEqual(Constants.EXPECT_TIC_TAC_TOE);
    });

    it("should render styles correctly", () => {
        expect(wrapper.find("div").at(0).hasClass(Constants.EXPECT_APP)).toBeTruthy();
        expect(wrapper.find("header").hasClass(Constants.EXPECT_APP_HEADER)).toBeTruthy();
    })
});