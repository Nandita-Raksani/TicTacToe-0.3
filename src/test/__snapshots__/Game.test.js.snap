// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Game/> component should render correctly 1`] = `ShallowWrapper {}`;
